### PR TITLE
[45243]  NSObject Binding methods should use NSString for the binding, not string

### PR DESCRIPTION
**Pull request by timrisi**

### CODE REVIEW

File: src/foundation.cs [hunk 1]
```diff
@@ -7468,32 +7468,62 @@ interface NSObject2 : NSObjectProtocol {
...
+		[Static]
+		[Obsolete ("Use SetDefaultPlaceholder (NSObject placeholder, NSObject marker, NSString binding) instead")]
+		[Wrap ("SetDefaultPlaceholder (placeholder, marker, (NSString) binding)")]
+		void SetDefaultPlaceholder (NSObject placeholder, NSObject marker, string binding);
+
+		[Export ("exposedBindings")]
+		NSString[] ExposedBindings ();
+#else
+		[Export ("exposedBindings")]
+		NSString[] ExposedBindings;
```
Comment (spouliot) [inline src/foundation.cs]: I think you're missing a `{ get; }` there

**Comment by timrisi** [comment]
Is there a way to keep the old versions and still bind the new ones to be available at the same time?

**Comment by spouliot** [comment]
For the API listed above (i.e. the `static` one are fine) you can rework this by using `[Sealed]` on the new API (and drop the `[Wrap]`). That will allow you to _re-use_ (kind of) the same `[Export]` for the `NSString` variant - they won't be virtual so they won't be registered (so the "use only once" rule does not apply).

IOW it's the new `NSString` overloads that will be non-virtual (for`!XAMCORE_4_0`). The main drawback is that it won't be possible to override them.

**Comment by spouliot** [comment]
@chamons macOS specific fix, please review too :)

**Comment by chamons** [comment]
@timrisi Assuming API diff says you didn't break anything 👍

**Comment by spouliot** [comment]
@timrisi please follow up on last comment, thanks!

**Comment by timrisi** [comment]
Namespace Foundation

Type Changed: Foundation.NSObject

Obsoleted methods:

	[Obsolete ("Use Bind (NSString binding, NSObject observable, string keyPath, [NullAllowed] NSDictionary options) instead")]
	public virtual void Bind (string binding, NSObject observable, string keyPath, NSDictionary options);
	[Obsolete ("Use GetBindingInfo (NSString binding) instead")]
	public virtual NSDictionary BindingInfo (string binding);
	[Obsolete ("Use GetBindingOptionDescriptions (NSString aBinding) instead")]
	public virtual NSObject[] BindingOptionDescriptions (string aBinding);
	[Obsolete ("Use GetBindingValueClass (NSString binding) instead")]
	public virtual ObjCRuntime.Class BindingValueClass (string binding);
	[Obsolete ("Use SetDefaultPlaceholder (NSObject placeholder, NSObject marker, NSString binding) instead")]
	public static void SetDefaultPlaceholder (NSObject placeholder, NSObject marker, string binding);
	[Obsolete ("Use Unbind (NSString binding) instead")]
	public virtual void Unbind (string binding);
Added methods:

	public void Bind (NSString binding, NSObject observable, string keyPath, NSDictionary options);
	public NSDictionary GetBindingInfo (NSString binding);
	public NSObject[] GetBindingOptionDescriptions (NSString aBinding);
	public ObjCRuntime.Class GetBindingValueClass (NSString binding);
	public static NSObject GetDefaultPlaceholder (NSObject marker, NSString binding);
	public static void SetDefaultPlaceholder (NSObject placeholder, NSObject marker, NSString binding);
	public void Unbind (NSString binding);

**Pull request merged**